### PR TITLE
Prototype signature files for lex files

### PR DESCRIPTION
**Pull request by nojaf**
See https://github.com/fsprojects/FsLexYacc/pull/170

Before:
```
--------------------------------------------------------------------------------------------------------
|Phase name                          |Elapsed |Duration| WS(MB)|  GC0  |  GC1  |  GC2  |Handles|Threads|
|------------------------------------|--------|--------|-------|-------|-------|-------|-------|-------|
|>Import mscorlib+FSharp.Core        |  0.1552|  0.1482|    116|      0|      0|      0|    282|     31|
|>Parse inputs                       |  0.5886|  0.4247|    641|      2|      1|      1|    424|     69|
|>Import non-system references       |  0.5931|  0.0017|    642|      0|      0|      0|    424|     69|
|>Typecheck                          |  4.4414|  3.8460|   3839|      5|      3|      1|    460|     70|
```

After:
```
--------------------------------------------------------------------------------------------------------
|Phase name                          |Elapsed |Duration| WS(MB)|  GC0  |  GC1  |  GC2  |Handles|Threads|
|------------------------------------|--------|--------|-------|-------|-------|-------|-------|-------|
|>Import mscorlib+FSharp.Core        |  0.1568|  0.1498|    116|      0|      0|      0|    282|     31|
|>Parse inputs                       |  0.6245|  0.4603|    655|      2|      1|      1|    426|     69|
|>Import non-system references       |  0.6293|  0.0014|    656|      0|      0|      0|    426|     69|
|>Typecheck                          |  4.0846|  3.4530|   4293|      4|      3|      1|    524|     70|
```

image

### CODE REVIEW

File: src/Compiler/FSharp.Compiler.Service.fsproj [hunk 1]
```diff
@@ -244,6 +247,9 @@
     <Compile Include="$(FsYaccOutputFolder)pppars.fs">
       <Link>SyntaxTree\FsYaccOutput\pppars.fs</Link>
     </Compile>
+<!--    <Compile Include="$(FsYaccOutputFolder)pars.fsi">-->
```
Comment (abonie) [inline src/Compiler/FSharp.Compiler.Service.fsproj:250]: Presumably this shouldn't be here (or should be uncommented?)

Comment (nojaf) [reply src/Compiler/FSharp.Compiler.Service.fsproj:250]: Ah, no that is from the other PR. Should be removed here.

**Review by abonie — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

File: buildtools/fslex/fslexdriver.fs [hunk 2]
```diff
@@ -147,56 +176,82 @@ let writeUnicodeTranslationArray dfaNodes domain (writer: Writer) =
...
+        writer.WriteLine "and %s %s lexbuf =" ident arguments
+        
+        let signature =
+            if List.isEmpty args then
+                sprintf "val %s: lexbuf: LexBuffer<char> -> token" ident
+            else
+                args
+                |> List.map (function
+                    | RuleArgument.Ident ident ->
+                        // This is not going to lead to a valid signature file, the only workaround is that the caller will specify the type.
```
Comment (T-Gro) [inline buildtools/fslex/fslexdriver.fs:224]: What does it mean in practise that it is not valid?
Do we have any issues with current code?

Comment (nojaf) [reply buildtools/fslex/fslexdriver.fs:224]: If you didn't add a type annotation in the `*.fsl` file, the generated signature file becomes invalid and you cannot use it in your project. The existing `*.fs` will not be affected by this.

**Review by edgarfgp — APPROVED** [review]

**Pull request merged**